### PR TITLE
gamemodes button now '""""work''"""""

### PR DESCRIPTION
**Pull request by lachlanshoesmith**
also changed some css styling that doesnt quite make sense with the cards
also changed how sheets handle their classname prop - i imagine there is a more eloquent way of doing it than the way i did tho so feel free to correct me i didnt google lol

### CODE REVIEW

File: frontend/src/components/Card/Card.module.scss [hunk 1]
```diff
@@ -7,13 +7,13 @@
...
-  cursor: pointer;
   transition: transform 0.3s;
   border-radius: 5px;
-}
-
-.card:hover {
-  transform: scale(1.04);
+  
+  &:hover {
+    cursor: pointer;
```
Comment (bengodw) [inline frontend/src/components/Card/Card.module.scss:14]: idk what the difference between these is but cool

File: frontend/src/components/Sheet/Sheet.module.scss [hunk 2]
```diff
@@ -16,8 +16,6 @@
   display: flex;
   flex-direction: row;
   justify-content: center;
-  z-index: -1;
```
Comment (bengodw) [inline frontend/src/components/Sheet/Sheet.module.scss:19]: these bits were to make sure the gamemodes sheet appears behind the mobile navbar - were they causing other problems

File: frontend/src/pages/GamemodesPage/GamemodesPage.tsx [hunk 3]
```diff
@@ -15,13 +17,15 @@ function GamemodesPage() {
           cardTitleText="Exploration"
           cardBodyText="Explore UNSW’s lush campus at your own pace!"
           img={poggers}
+          onClick={() => { navigate('/play'), { state: { gamemode: 'exploration' } } }}
```
Comment (bengodw) [inline frontend/src/pages/GamemodesPage/GamemodesPage.tsx]: confused about how this passes the state through

File: frontend/src/components/Card/Card.module.scss (hunk 1 above)
Comment (lachlanshoesmith) [reply frontend/src/components/Card/Card.module.scss:14]: just sass things

File: frontend/src/components/Sheet/Sheet.module.scss (hunk 2 above)
Comment (lachlanshoesmith) [reply frontend/src/components/Sheet/Sheet.module.scss:19]: yeah hover didnt work at all

File: frontend/src/pages/GamemodesPage/GamemodesPage.tsx (hunk 3 above)
Comment (lachlanshoesmith) [reply frontend/src/pages/GamemodesPage/GamemodesPage.tsx]: https://reactrouter.com/en/main/hooks/use-navigate#optionsstate

Comment (lachlanshoesmith) [reply frontend/src/pages/GamemodesPage/GamemodesPage.tsx]: i fucked up the syntax tho ill fix

Comment (lachlanshoesmith) [reply frontend/src/pages/GamemodesPage/GamemodesPage.tsx]: > i fucked up the syntax tho ill fix

fixed in latest commit

File: frontend/src/components/Sheet/Sheet.module.scss (hunk 2 above)
Comment (lachlanshoesmith) [reply frontend/src/components/Sheet/Sheet.module.scss:19]: reapplying these properties doesnt seem to change anything on my end unless im missing smth? maybe my screen is too big?

image
image

Comment (lachlanshoesmith) [reply frontend/src/components/Sheet/Sheet.module.scss:19]: doesnt seem to do much on iphone xr
image
image

File: frontend/src/pages/GamemodesPage/GamemodesPage.tsx (hunk 3 above)
Comment (lachlanshoesmith) [reply frontend/src/pages/GamemodesPage/GamemodesPage.tsx]: made changes in `game` branch to retrieve gamemode from this, ill resolve for now but pls msg if i can clarify at all

File: frontend/src/components/Sheet/Sheet.module.scss (hunk 2 above)
Comment (bengodw) [reply frontend/src/components/Sheet/Sheet.module.scss:19]: Screenshot 2024-09-13 at 5 25 44 pm
Screenshot 2024-09-13 at 5 25 49 pm

sorry I should've been clearer this is the difference

**Review by bengodw — COMMENTED** [review]
should change back to z-level / relative unless you have better fix

**Pull request merged**